File: library/src/main/java/com/github/pelmenstar1/rangecalendar/LocaleUtils.java
package com.github.pelmenstar1.rangecalendar;

import android.content.Context;
import android.content.res.Configuration;
import android.content.res.Resources;
import android.os.Build;

import org.jetbrains.annotations.NotNull;

import java.util.Locale;

final class LocaleUtils {
    @NotNull
    public static Locale getLocale(@NotNull Context context) {
        Resources res = context.getResources();
        Configuration config = res.getConfiguration();

        if(Build.VERSION.SDK_INT >= 24) {
            return config.getLocales().get(0);
        } else {
            //noinspection deprecation
            return config.locale;
        }
    }
}

File: library/src/main/java/io/github/pelmenstar1/rangecalendar/LocaleUtils.kt
package io.github.pelmenstar1.rangecalendar

import android.content.Context
import android.os.Build
import java.util.*

fun Context.getLocaleCompat(): Locale {
    val config = resources.configuration

    return if (Build.VERSION.SDK_INT >= 24) {
        config.locales[0]
    } else {
        @Suppress("DEPRECATION")
        config.locale
    }
}